src/recipe/admin/src/runtime/components/DataTable.vue

```diff
@@ -52,8 +88,16 @@ const { schema, data } = defineProps({
   data: { type: null, required: true }
 });
 
-const { columns, dataKey, paginator, rows, rowsWhenLoading } = schema;
+const { columns, dataKey, footer, paginator, rows, rowsWhenLoading, scrollHeight } = schema;
 
 const loading = context.loading();
-const value = computed(() => data ?? new Array(rowsWhenLoading || 5).fill({ }));
+// accept has footer when schema includes `footer.columns` and
+// data has `items`
+const hasFooterData = computed(() => footer?.columns && data?.items);
+// return `data.items` when has footer, else use data as array
+const dataItems = computed(() => hasFooterData.value ? data.items : data);
+const value = computed(() => data ? dataItems.value : new Array(dataItems?.value?.length || rowsWhenLoading || 5).fill({ }));
```

Comment:
`dataItems?.value?.length` is unnecessary here, it can't be not null in this else condition